apps/buildog/src/components/org-navigation.tsx

```diff
@@ -1,66 +1,219 @@
+                />
+                <AvatarFallback>{selectedAccount.name.charAt(0)}</AvatarFallback>
+              </Avatar>
+              <div className="ml-1 mr-auto">{selectedAccount.name}</div>
+            </>
+          ) : (
+            "Select Organization"
+          )}
+          <CaretSortIcon className="ml-2 h-4 w-4 shrink-0 opacity-50" />
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent className="w-[200px] p-0">
+        <Command shouldFilter={false}>
+          <CommandInput
+            placeholder="Search organization..."
+            value={searchTerm}
+            onValueChange={setSearchTerm}
+          />
+          <CommandList>
+            <CommandEmpty>No account or organization found.</CommandEmpty>
+            {filteredPersonalAccounts.length > 0 && (
+              <CommandGroup>
+                <div className="pl-3 pr-3 font-medium text-zinc-400 text-xs pt-2 pb-2">
+                  Personal Account
+                </div>
+                {filteredPersonalAccounts.map((account) => (
+                  <CommandItem
+                    key={account.id}
+                    onSelect={() => {
+                      setSelectedAccount(account);
+                      setSelectedOrg(null);
+                      setValue(account.id.toString() === value ? "" : account.id.toString());
+                      setOpen(false);
+                    }}
+                  >
+                    <Avatar className="relative flex shrink-0 overflow-hidden rounded-full mr-2 h-5 w-5">
+                      <AvatarImage
+                        className="aspect-square h-full w-full grayscale"
+                        src={account.avatarUrl}
+                        alt={account.name}
+                      />
+                      <AvatarFallback>{account.name.charAt(0)}</AvatarFallback>
+                    </Avatar>
+                    {account.name}
+                    <CheckIcon
+                      className={cn(
+                        "mr-0 ml-auto h-4 w-4",
+                        selectedAccount?.id === account.id ? "opacity-100" : "opacity-0"
+                      )}
+                    />
+                  </CommandItem>
+                ))}
+              </CommandGroup>
+            )}
+            {filteredOrganizations.length > 0 && (
+              <CommandGroup>
+                <div className="pl-3 pr-3 font-medium text-zinc-400 text-xs pt-2 pb-2">
+                  Organizations
+                </div>
+                {filteredOrganizations.map((org) => (
+                  <CommandItem
+                    key={org.OrganizationId}
+                    value={org.OrganizationId.toString()}
+                    onSelect={() => {
+                      setSelectedOrg(org);
+                      setSelectedAccount(null);
+                      setValue(
+                        org.OrganizationId.toString() === value ? "" : org.OrganizationId.toString()
+                      );
```

Comment:
1. org.OrganizationId.toString() -> org.OrganizationId is string, you don't need to manipulate type here

2. if org.OrganizationId is same as selected value, we don't need to switch value to empty string. We don't need this check `org.OrganizationId.toString() === value ? "" : org.OrganizationId.toString()` _(If there is a reason for this, let's discuss on discord)_